src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.utility.Parser;
+import duke.utility.Storage;
+import duke.utility.TaskList;
+import duke.utility.Ui;
+
+/**
+ * Class to represents the Duke.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a new Duke object to initialized the program.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage("data/storage.txt");
+        try {
+            tasks = new TaskList(storage.loadTasksFromDisk());
+        } catch (DukeException e) {
+            ui.showError(e);
+            tasks = new TaskList();
+        }
+    }
+
+    public String getResponse(String input) {
+        try {
+            Command command = Parser.parseUserInput(input);
+            return command.execute(tasks, ui, storage);
+        } catch (DukeException e) {
+            return e.getMessage();
+        }
+    }
+
+    private void takeUserInput() {
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String userInput = ui.readLine();
+                Command command = Parser.parseUserInput(userInput);
+                command.execute(tasks, ui, storage);
+                isExit = command.isExit();
+            } catch (DukeException e) {
+                ui.showError(e);
+            }
+        }
+    }
+
+    private void run() {
+        ui.welcome();
+        takeUserInput();
+        ui.exit();
+    }
+
+    /**
+     * Run the Duke program.
+     * @param args The command line input
+     */
```

Comment:
Perhaps start with "Runs" instead of "Run" as specified in the coding standard. There should also be a empty line between description and parameter section. Punctuation behind each parameter description can also be added to follow the JavaDoc comments coding standard form.